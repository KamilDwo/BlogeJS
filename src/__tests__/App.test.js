import React from 'react'
import ReactDOM from 'react-dom'
import StoredApp from '../App'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import store from '../store/store'
import { Provider } from 'react-redux'
import toJson from 'enzyme-to-json'

const options = {
  disableLifecycleMethods: true
}

configure({ adapter: new Adapter() })

describe('<StoredApp/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<Provider store={store}><StoredApp/></Provider>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches to snapschot', () => {
    const wrapper = shallow(<Provider store={store}><StoredApp/></Provider>, options).dive({ context: { store } }).dive().dive();

    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
