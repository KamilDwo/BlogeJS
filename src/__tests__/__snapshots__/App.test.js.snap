// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoredApp/> matches to snapschot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Adapter>
    <GlobalStyleComponent
      suppressMultiMountWarning={false}
    />
    <BackTop
      visibilityHeight={400}
    />
    <Styled(Sider)
      breakpoint="lg"
      collapsedWidth="50"
      collapsible={true}
      onCollapse={[Function]}
    >
      <styled.div
        collapsed=""
      >
        BlogeJS
      </styled.div>
      <Styled(Menu)
        collapsed=""
        mode="inline"
        selectedKeys={
          Array [
            "1",
          ]
        }
        theme="dark"
      >
        <Styled(MenuItem)
          key="1"
        >
          <Link
            replace={false}
            to="/"
          >
            <Icon
              theme="outlined"
              type="home"
            />
            <span
              className="nav-text"
            >
              Latest
            </span>
          </Link>
        </Styled(MenuItem)>
        <Styled(MenuItem)
          key="2"
        >
          <Link
            replace={false}
            to="/bloggers"
          >
            <Icon
              theme="outlined"
              type="team"
            />
            <span
              className="nav-text"
            >
              Bloggers
            </span>
          </Link>
        </Styled(MenuItem)>
        <Styled(MenuItem)
          key="3"
        >
          <Link
            replace={false}
            to="/beloved"
          >
            <Icon
              theme="outlined"
              type="heart"
            />
            <span
              className="nav-text"
            >
              Beloved
            </span>
          </Link>
        </Styled(MenuItem)>
        <Styled(MenuItem)
          key="4"
        >
          <div
            onClick={[Function]}
          >
            <Icon
              type="user"
            />
            <span
              className="nav-text"
            >
              Userpanel
            </span>
          </div>
        </Styled(MenuItem)>
      </Styled(Menu)>
    </Styled(Sider)>
    <Styled(Adapter)
      collapsed=""
    >
      <Adapter
        style={
          Object {
            "margin": "16px 16px 0",
          }
        }
      >
        <ContextProvider
          value={
            Object {
              "loggedUser": false,
            }
          }
        >
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/bloggers"
            />
            <Route
              component={[Function]}
              path="/beloved"
            />
            <Route
              component={[Function]}
              path="/user"
            />
            <Route
              component={[Function]}
              path="/post/:id"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </ContextProvider>
      </Adapter>
      <Adapter
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <Foot />
      </Adapter>
    </Styled(Adapter)>
  </Adapter>
</Router>
`;
